### ISSUE TITLE
`global` is not defined on Vite

### ISSUE THREAD

**Issue by antfu** [nuxt/framework#1862]
### Environment

The `test/fixture/bridge`

------------------------------
- Operating System: `Darwin`
- Node Version:     `v16.11.1`
- Nuxt Version:     `2.15.8`
- Package Manager:  `yarn@3.1.0`
- Bundler:          `Webpack`
- User Config:      `components`, `serverMiddleware`, `buildDir`, `plugins`, `nitro`, `bridge`, `buildModules`
- Runtime Modules:  `-`
- Build Modules:    `@nuxt/bridge@3.0.0`
------------------------------

### Describe the bug

```bash
cd test/fixtures/bridge
TEST_BRIDGE_VITE=1 yarn dev
```

Visit the app, and get the error in the console

![image](https://user-images.githubusercontent.com/11247099/141299655-31e8ea73-b45c-4652-a879-0fa77b6cc490.png)

It comes from Nuxt 2's entry file.

### Reproduction

-

### Additional context

I guess a quick solution is injecting this on dev:

```ts
if (typeof global === 'undefined')
  globalThis.global = globalThis
```

**Comment by tobiasdiez** [comment]
This is now fixed for me.

**Comment by pi0** [comment]
Only use by Nuxt internals resolved by (#1866). Moving enhancement to prevent this issue to #1922
